gensim/models/word2vec.py

```diff
@@ -748,7 +748,7 @@ def most_similar(self, positive=[], negative=[], topn=10):
         dists = dot(self.syn0norm, mean)
         if not topn:
             return dists
-        best = argsort(dists)[::-1][:topn + len(all_words)]
+        best = matutils.argsort(dists, topn + len(all_words))[::-1][:topn + len(all_words)]
```

Comment:
`matutils.argsort` already returns the greatest elements... the logic here looks too convoluted.